src/format.rs

```diff
@@ -169,11 +174,19 @@ impl<T: Display> LogFormat<(u32, StructuredData, T)> for Formatter5424 {
     ) -> Result<()> {
         let (message_id, data, message) = log_message;
 
+        // Guard against sub-second precision over 6 digits per rfc5424 section 6
+        let timestamp = time::OffsetDateTime::now_utc();
+        // SAFETY: timestamp range is enforced, so this will never fail
+        let timestamp = timestamp
+            // Removing significant figures beyond 6 digits (nanoseconds to microseconds)
+            .replace_nanosecond(timestamp.microsecond())
```

Comment:
I am not sure this will give a correct result. If you have a timestamp like `1985-04-12T23:20:50.123456789Z`, then by replacing the nanoseconds `123 456 789` with the microseconds `123 456` then we would get `1985-04-12T23:20:50.123123456Z`

The [comment on the issue](https://github.com/Geal/rust-syslog/issues/73#issuecomment-1548047732) gives a solution that should work (assuming the RFC3339 formatter will always elide trailing zeroes), but a more robust approach would write a custom formatter: https://docs.rs/time/latest/time/macros/macro.format_description.html

Reply:
Switched to that solution

> assuming the RFC3339 formatter will always elide trailing zeroes

time does https://github.com/time-rs/time/blob/38beb4a2e9e8eacfceef413c3a54fd56800510a7/time/src/formatting/formattable.rs#L250-L268

And chrono can also limit precision via:
```
chrono::Utc::now().to_rfc3339_opts(chrono::SecondsFormat::Micros, true)
```

Reply:
alright